### DISCUSSION TITLE
Loading content at runtime?

### DISCUSSION THREAD

**Discussion by jjb-pro** [stride3d/stride#1665] [General]
**Hello,
I'm still a Unity user, but I'm thinking of switching to Stride. I am planning a game that allows add-ons/mods.** 

Is it possible to do the loading of textures, lights, models, flares, audio while running the game (and not before) and do this in other threads? I have my own file structure for the mods. 

**Thanks in advance for the help!**

**Comment by NicusorN5** [comment]
Yes, it is. There are few methods:
-> Running Stride's content compiler in your game. Unfortunately you won't be able to compile the game for Linux and MacOS.
-> Manually load and process your assets by using assimp(models) and DirectXTex/GLI(images). This is very hard to implement, and you'll have to work with Stride's source code.
-> Provide modding tools that use Stride's asset compiler for use in your own game. Just like Source engine has tools for managing resources. 
-> Provide the players/ modders a game API w/ Stride, the same way you'd create modifications for Unreal Engine games.

**Reply by jjb-pro** [reply]
Some of your suggestions are unfortunately useless for me, because I use my own file formats and cannot change them. Anyway, thank you for the answer!

**Reply by manio143** [reply]
> Some of your suggestions are unfortunately useless for me, because I use my own file formats and cannot change them.

If you're using your own file formats you will need to write a converter from that into the Stride's internal image/model representation. It's definitely doable.

**Reply by jjb-pro** [reply]
Thanks for the answer! Then I'll download Stride and try it out!
